### ISSUE TITLE
Method to determine if a given key already has a timer set

### ISSUE THREAD

**Issue by jonhoo** [andrewjstone/ferris#5]
Currently there is no way to check if a timer already exists for a key (and if it does, how long remains). This would be really handy for debugging, even though it probably has limited use-cases for production code.

**Comment by andrewjstone** [comment]
I think it's easy to provide the first part (checking if the key exists) by doing a lookup in the cancel set. However, the second part is tricky without providing significant overhead. I believe you'd either need to check wall clock time at start and store it with the key in a cancel hashmap and then compare against current time, or do an exhaustive search through the wheels to figure out where it is in relation to the current pointers. I'm a bit reluctant to make this change for perf reasons.

**Comment by andrewjstone** [comment]
I'm going to go ahead and close this as it doesn't really fit with the design of this project. This way I can focus on implementing the other issues that we agree need fixing. If you think of any way around what I wrote, feel free to reopen.

**Comment by jonhoo** [comment]
I think providing just the first (is there already a timer?) would still be useful in isolation.

**Comment by andrewjstone** [comment]
Ok doke. I will go ahead and do that.

**Comment by andrewjstone** [comment]
I suppose the latte exhaustive search could also be fine if done during debugging. I just don't want to hinder the production path by storing absolute timestamps.

**Comment by jonhoo** [comment]
Yeah, I agree with that. I think I'd just leave timestamps out of this particular thing, and *just* give efficient "check if timer for key exists" for now.
